### ISSUE TITLE
Add HTML5 navigator.geolocation options as input to the Geolocate control

### ISSUE THREAD

**Issue by andreipopovici** [mapbox/mapbox-gl-js#3473]
## Motivation

For mobile uses of Mapbox GL, developers would appreciate the enableHighAccuracy option to enable the device's GPS.
## Design Alternatives

In https://github.com/mapbox/mapbox-gl-js/blob/master/js/ui/control/geolocate_control.js#L8, the options to be passed to `navigator.geolocation.getCurrentPosition()` are hard-coded.

This object could be made a property of the `options` object passed to the Geolocate control before adding it to the map UI.
## Design

Advantage: finer-grained control for advanced use, with sensible (i.e. current) defaults.
### Implementation

``` javascript
class GeolocateControl extends Control {

    constructor(options) {
        super();
        this._geoOptions = options && options.geoOptions || { enableHighAccuracy: false, timeout: 6000 /* 6sec */ }
        this._position = options && options.position || 'top-right';
    }

[...]

window.navigator.geolocation.getCurrentPosition(this._success.bind(this), this._error.bind(this), this._geoOptions);
```

**Comment by mollymerp** [comment]
@andreipopovici thank you for submitting your issue! we would happily consider a pull request implementing this feature.

**Comment by andrewharvey** [comment]
hey @andreipopovici I like this idea, are you planning on implementing this as a pull request? If you are great, but if not I'll give this a go. I just don't want to step on your toes if you're working on it.

**Comment by andreipopovici** [comment]
Hi there,
By all means, have at it! :D I did not have time to clean the code I used for my needs, and it would take more to go back through it than for someone knowledgeable to replicate the (small) effort.

On Sat, Dec 3, 2016 at 12:24 PM, Andrew Harvey <notifications@github.com> wrote:
hey @andreipopovici [https://github.com/andreipopovici] I like this idea, are you planning on implementing this as a pull request? If you are great, but if not I'll give this a go. I just don't want to step on your toes if you're working on it.
